packages/core/src/render3/instructions/shared.ts

```diff
@@ -266,6 +268,19 @@ export function createTNodeAtIndex(
   return tNode;
 }
 
+/**
+ * Checks if the current component is declared inside of a standalone component template.
+ *
+ * @param lView An `LView` that represents a current component that is being rendered.
+ */
+export function isHostComponentStandalone(lView: LView): boolean {
+  !ngDevMode && throwError('Must never be called in production mode');
+
+  const declarationLView = lView[DECLARATION_COMPONENT_VIEW] as LView<Type<unknown>>;
+  const context = declarationLView[CONTEXT];
+  const componentDef = getComponentDef(context.constructor);
+  return !!(componentDef?.standalone);
```

Comment:
@dario-piotrowicz it looks like some tests are failing on CI because the `context` becomes undefined in some cases. Looking at some of the tests in the `render3` folder, they seem to use old testing infrastructure and we didn't have a chance to convert them to the new one yet. However, according to the docs for the `lView[CONTEXT]` field:
https://github.com/angular/angular/blob/0e14df697ad00a094b3a2a07e6e5e4295e978830/packages/core/src/render3/interfaces/view.ts#L176-L183
we should account for the fact that the `lView[CONTEXT]` might be `null` or `undefined`.

So I'd propose adding a check like this:
```suggestion
  const context = declarationLView[CONTEXT];
  
  // Unable to obtain a context, fall back to the non-standalone scenario.
  if (!context) return false;
  
  const componentDef = getComponentDef(context.constructor);
  return !!(componentDef?.standalone);
```

Reply:
sounds good, thanks a lot! :slightly_smiling_face: